boards/arm/nrf52840_pca20035/Kconfig.board

```diff
@@ -0,0 +1,17 @@
+# Kconfig - nRF52840 PCA20035 board configuration
+#
+# Copyright (c) 2019 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+
+if SOC_NRF52840_QIAA
+
+config BOARD_NRF52840_PCA20035_V0_2_2
```

Comment:
There is a slight problem with this: these appear to the user a `Kconfig` choice, and both are selectable. However, selecting anything else than the default will not have the expected result, since the dts will remain unchanged. It will possibly break stuff.

I stumbled on a similar issue here: https://github.com/NordicPlayground/fw-nrfconnect-zephyr/blob/master/boards/arm/nrf9160_pca10090/Kconfig.board.

What I did was to show one choice only when the other was not selected; that way only the correct choice is displayed. Perhaps we could do something similar here?

Reply:
Good catch, didn't think of that. Only 1 of 4 possible options in the choice are valid depending on the dts files that were selected, so having it in a choice doesn't really make much sense, but we'll have to stick to Zephyr convention here I guess. The straight-forward solution would be something like this:
```
if BOARD_NRF9160_PCA20035_V0_3_0

config BOARD_NRF9160_PCA20035_V0_2_2
	bool "nRF9160 PCA20035 v0.2.2"
	depends on !TRUSTED_EXECUTION_NONSECURE
	select BOARD_NRF9160_PCA2003

config BOARD_NRF9160_PCA20035_V0_2_2NS
	bool "nRF9160 PCA20035 v0.2.2 non-secure"
	depends on TRUSTED_EXECUTION_NONSECURE
	select BOARD_NRF9160_PCA20035NS

endif # !BOARD_NRF9160_PCA20035_V0_3_0

#if !BOARD_NRF9160_PCA20035_V0_2_2

config BOARD_NRF9160_PCA20035_V0_3_0
	bool "nRF9160 PCA20035 v0.3.0"
	depends on !TRUSTED_EXECUTION_NONSECURE
	select BOARD_NRF9160_PCA20035

config BOARD_NRF9160_PCA20035_V0_3_0NS
	bool "nRF9160 PCA20035 v0.3.0 non-secure"
	depends on TRUSTED_EXECUTION_NONSECURE
	select BOARD_NRF9160_PCA20035NS

endif # !BOARD_NRF9160_PCA20035_V0_2_2
```

But that's not possible because it causes a dependency loop. So I'm a bit stuck here.
Should I rather move it into the Kconfig file instead and make the options regular configs so that I avoid the dependency loop, but at the cost that it would then not show up in the board selection choice in menuconfig?

Reply:
I see your problem. I had fixed the issue with the board choice for `nrf9160_pca10090` upstream, but introducing a _version_ of a board complicates it further, indeed. I am working on this.

Reply:
Alright, I might have an idea.

Instead of providing both boards out of the box, we could approach it similarly to what I have done for `nrf52840_pca10059`. Basically, we provide all the necessary `dts` and `Kconfig` files for `nrf9160_pca20035` version 0.3.0 and 0.2.2 but only let the build system see  one board at a time. This would be done by commenting out a line in a couple of files, in the `dts` and `Kconfig.board` files which source the correct files for a given version.

Not sure if I explained myself well enough, but I have pushed a commit [here](https://github.com/lemrey/fw-nrfconnect-nrf/commit/0c2e6fdc56fffdfeb6d5ef729951cc4b23a6ec3f) so you can have a look.

Reply:
Thanks, I'll add a comment in the commit